### PR TITLE
Add withdraw task

### LINKED ISSUE #72: Add task for `withdraw`
We need to add a task for `withdraw` in the scenario player. https://github.com/raiden-network/raiden/blob/develop/raiden/api/rest.py#L1054

### PR DESCRIPTION
**Pull request by ulope**
Fixes #72

### CODE REVIEW

**Comment by ulope** [comment]
I think codecov is drunk...

File: tests/unittests/tasks/test_channels.py [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+        "task_name",
+        "task_params",
+        "expected_exception",
+        "expected_exception_message",
+        "expected_req_method",
+        "expected_req_url",
+        "expected_req_body",
+        "resp_code",
+        "resp_json",
+    ),
```
Comment (nlsdfnbch) [inline tests/unittests/tasks/test_channels.py]: A general pointer for consistency: I state `argnames` as a string in tests, and pass `argvalues` as a keyword argument. I think it helps reading the parametrize mark.

File: tests/unittests/tasks/test_channels.py [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+            'Expected all channels to be in "open" state.',
+            "GET",
+            f"http://0/api/v1/channels/0x1{1:039d}",
+            {},
+            200,
+            [{"state": "open"}, {"state": "closed"}],
+            id="assert_sum-state-mismatch",
+        ),
+    ],
+)
```
Comment (nlsdfnbch) [inline tests/unittests/tasks/test_channels.py]: Holy moly, that's a lot of parameters! Just an observation, carry on.

File: tests/unittests/tasks/test_channels.py [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+        assert response == resp_json
+
+    if expected_req_method:
+        assert mocked_responses.calls[0].request.method == expected_req_method
+        assert mocked_responses.calls[0].request.url == expected_req_url
+        assert mocked_responses.calls[0].request.body.decode() == json.dumps(expected_req_body)
+        assert mocked_responses.calls[0].response.json() == resp_json
+
+
+def test_store_channel_info(mocked_responses, dummy_scenario_runner, api_task_by_name):
```
Comment (nlsdfnbch) [inline tests/unittests/tasks/test_channels.py]: This should be two tests. 

```
def test_store_channel_info_raises_exception_on_missing_key_config():
    ...

def test_store_channel_info_whatever_youre_testing_in_the_second_part():
     ...
```

File: tests/unittests/tasks/test_channels.py [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+        "GET",
+        f"http://0/api/v1/channels/0x1{1:039d}/0x2{1:039d}",
+        json={"something": 1},
+        status=200,
+    )
+    task_instance = api_task_by_name("store_channel_info", {"from": 0, "to": 1, "key": "test"})
+
+    task_instance()
+
+    assert dummy_scenario_runner.task_storage[STORAGE_KEY_CHANNEL_INFO]["test"] == {"something": 1}
```
Comment (nlsdfnbch) [inline tests/unittests/tasks/test_channels.py]: From reading this, I'm not entirely sure what you're testing here - adding a comment or similar would help.

File: tests/unittests/tasks/test_channels.py (hunk 1 above)
Comment (ulope) [reply tests/unittests/tasks/test_channels.py]: I'm not a fan of mangling the argnames into a single string. That's why we have tuples ;)

Also I'd argue we should be using the same code style here as everywhere else in raiden.

Comment (nlsdfnbch) [reply tests/unittests/tasks/test_channels.py]: > Also I'd argue we should be using the same code style here as everywhere else in raiden.

Yea, that's a pretty good argument.

File: scenario_player/tasks/channels.py [hunk 5]
```diff
@@ -67,6 +67,14 @@ def _request_params(self):
         return dict(total_deposit=self._config["total_deposit"])
 
 
+class WithdrawTask(ChannelActionTask):
+    _name = "withdraw"
```
Comment (nlsdfnbch) [inline scenario_player/tasks/channels.py]: Oh no, a task class without doc strings :<

File: tests/unittests/tasks/test_channels.py [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+    because it raises an exception before getting to the request).
+
+    If ``expected_exception`` (and optional ``expected_exception_message``) is given we assert
+    this exception is actually raised.
+
+    Combinations of both of the above behaviours allow to assert on these three cases:
+      - Successful task execution
+      - Exception raised before a request is performed (e.g. violated precondition)
+      - Exception raised after a request is performed (e.g. processing return value)
+    """
```
Comment (nlsdfnbch) [inline tests/unittests/tasks/test_channels.py]: mmmh, delicious docs. :+1:

**Comment by czepluch** [comment]
Would be nice if you could also add an entry in the example file.

**Pull request merged**